tests/callbacks/test_stochastic_weight_avg.py

```diff
@@ -281,3 +307,89 @@ def on_train_epoch_start(self):
     )
     trainer.fit(model)
     assert model.on_train_epoch_start_called
+
+
+def _swa_resume_training_from_checkpoint(tmpdir, crash_after_epoch=4, ddp=False):
+    model = SwaTestModel(crash_after_epoch=crash_after_epoch)
+    swa_start = 3
+    max_epochs = 5
+    swa_callback = SwaTestCallback(swa_epoch_start=swa_start, swa_lrs=0.1)
+
+    num_processes = 2 if ddp else 1
+    strategy = "ddp_spawn_find_unused_parameters_false" if ddp else None
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        enable_progress_bar=False,
+        max_epochs=max_epochs,
+        limit_train_batches=5,
+        limit_val_batches=0,
+        callbacks=[swa_callback],
+        accumulate_grad_batches=2,
+        num_processes=num_processes,
+        strategy=strategy,
+    )
+
+    exception_type = Exception if ddp else RuntimeError
+    backward_patch = mock.patch.object(Strategy, "backward", wraps=trainer.strategy.backward)
+    with backward_patch, pytest.raises(exception_type):
+        trainer.fit(model)
+
+    checkpoint_dir = Path(tmpdir) / "lightning_logs" / "version_0" / "checkpoints"
```

Comment:
```suggestion
    checkpoint_dir = trainer.checkpoint_callback.dirpath
```

Reply:
I tried making this change but this caused the `test_swa_resume_training_from_checkpoint_ddp` test to fail as for some reason `trainer.checkpoint_callback.dirpath` is None when using the `ddp_spawn_find_unused_parameters_false` strategy.

Reply:
If I debug this, I can see `__resolve_ckpt_dir` being called on an instance of `ModelCheckpoint`, but this happens in a spawned process and the instance is different to the one referenced by `trainer.checkpoint_callback` in the test method.

Reply:
hmm. weird